cli/popper/commands/cmd_run.py

```diff
@@ -110,6 +108,12 @@ def run_pipeline(project_root, pipeline, timeout, skip):
             for t in ['.err', '.out']:
                 with open('popper_logs/{}{}'.format(stage_file, t), 'r') as f:
                     pu.info(f.read())
+
+            if stage != 'teardown' and 'teardown' in pipeline['stages'] and \
+               pu.get_filename(abs_path, 'teardown') and \
+               (not skip or 'teardown' not in skip.split(',')):
+                execute(pu.get_filename(abs_path, 'teardown'), timeout)
+
```

Comment:
The new utility function is already checking if the file exist, so you could do:

```python
teardown_file = pu.get_filename(abs_pat, 'teardown')

if (teardown_file and
      stage != 'teardown' and
      'teardown' in pipeline['stages'] and
      'teardown' not in skipped):
    execute(teardown_file, timeout)
```

where `skipped` would be obtained in the way mentioned on the other comment